packages/nightingale-links/src/nightingale-links.ts

```diff
@@ -0,0 +1,231 @@
+    ) {
+      if (this._rawData) {
+        this.#contacts = getContactsObject(
+          filterContacts(this._rawData, this.minDistance, this.minProbability)
+        );
+        this.createTrack();
+      }
+    }
+  }
+
+  set contacts(data: LinksData) {
+    if (typeof data === "string") {
+      this._rawData = parseToRowData(data);
+    } else if (Array.isArray(data)) {
+      this._rawData = data;
+    } else {
+      throw new Error("data is not in a valid format");
+    }
+    this.#contacts = getContactsObject(
+      filterContacts(this._rawData, this.minDistance, this.minProbability)
+    );
+    this.createTrack();
+  }
+
+  protected createTrack() {
+    if (!this.#contacts) {
+      return;
+    }
+    // this.layoutObj?.init(this.#data);
+
+    this.svg?.selectAll("g").remove();
+
+    this.svg = select(this as unknown as NightingaleElement)
+      .selectAll<SVGSVGElement, unknown>("svg")
+      .attr("width", this.width)
+      .attr("height", this.height);
+
+    if (!this.svg) return;
+    this.seqG = this.svg.append("g").attr("class", "sequence-features");
+    this.createFeatures();
+    // this.#highlighted = this.svg.append("g").attr("class", "highlighted");
+    this.margins = this.svg.append("g").attr("class", "margin");
+  }
+
+  _getColor(d: number): string {
+    if (!this.#contacts?.contacts[d]) return "";
+    return d3Color(
+      this.#contacts.contacts[d].size / this.#contacts.maxNumberOfContacts
+    );
+  }
+
+  _dispatchSelectNode(d: number): void {
+    if (!this.#contacts) return;
+    this.#contacts.selected = d;
+    this.dispatchEvent(
+      getHighlightEvent(
+        "mouseover",
+        this,
+        Array.from(this.#contacts.contacts[d])
+          .concat(+d)
+          .sort()
+      )
+    );
+  }
+
+  createFeatures(): void {
+    if (!this.#contacts) return;
+    // this.removeEventListener("click", this._resetEventHandler);
+
+    this.seqG?.selectAll("g.contact-group").remove();
+    const contactGroup = this.seqG?.append("g").attr("class", "contact-group");
+    this.seqG?.append("g").attr("class", "links-group");
+
+    if (contactGroup)
+      this.contactPoints = contactGroup
+        .selectAll(".contact-point")
+        .data(Object.keys(this.#contacts.contacts).map(Number))
+        .enter()
+        .append("circle")
+        .attr("class", "contact-point")
+        .attr("fill", (d: number) => this._getColor(d))
+        .attr("id", (d: number) => `cp_${d}`)
+        .style("stroke-width", 2)
+        .on("mouseover", (_: Event, d: number) => {
+          if (this.#contacts?.isHold) return;
+          this._dispatchSelectNode(d);
+          this.refresh();
+        })
+        .on("mouseout", () => {
+          if (!this.#contacts || this.#contacts?.isHold) return;
+          this.#contacts.selected = undefined;
+          this.dispatchEvent(getHighlightEvent("mouseout", this));
+          this.refresh();
+        })
+        .on("click", (d: number) => {
```

Comment:
```suggestion
        .on("click", (_: Event, d: number) => {
```

Otherwise `d` is the click event, and when passed to `_dispatchSelectNode`, `this.#contacts.contacts[d]` is `undefined`.